pkg/operator/latestrevisionclient.go

```diff
@@ -0,0 +1,65 @@
+package operator
+
+import (
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/library-go/pkg/operator/revisioncontroller"
+)
+
+// OpenshiftDeploymentLatestRevisionClient is an LatestRevisionClient implementation for openshift-apiserver..
+type OpenshiftDeploymentLatestRevisionClient struct {
+	v1helpers.OperatorClient
+	TypedClient operatorv1client.OpenShiftAPIServersGetter
+}
+
+var _ revisioncontroller.LatestRevisionClient = OpenshiftDeploymentLatestRevisionClient{}
+
+func (c OpenshiftDeploymentLatestRevisionClient) GetLatestRevisionState() (*operatorv1.OperatorSpec, *operatorv1.OperatorStatus, int32, string, error) {
+	o, err := c.TypedClient.OpenShiftAPIServers().Get("cluster", metav1.GetOptions{})
```

Comment:
ah, so we will take the revision from the operator.

Reply:
Yes, the same way the static pod variant works, just different operator config types. And as we know the types, I just use the typed client here instead of dynamic client magic.